collect_app/src/androidTest/java/org/odk/collect/android/feature/instancemanagement/AutoSendTest.kt

```diff
@@ -205,4 +212,37 @@ class AutoSendTest {
 
         mainMenuPage.assertNumberOfFinalizedForms(1)
     }
+
+    @Test
+    fun whenFormHasAutoSend_formsAreSentInOldestFirstOrder() {
+        rule.startAtMainMenu()
+            .setServer(testDependencies.server.url)
+            .copyForm("one-question-autosend.xml")
+
+            .startBlankForm("One Question Autosend")
+            .inputText("31")
+            .swipeToEndScreen()
+            .clickSend()
+
+            .startBlankForm("One Question Autosend")
+            .inputText("32")
+            .swipeToEndScreen()
+            .clickSend()
+
+        testDependencies.scheduler.runDeferredTasks()
+
+        val root1 = parseXml(testDependencies.server.submissions[0]).rootElement
+        val root2 = parseXml(testDependencies.server.submissions[1]).rootElement
+
+        assertThat((root1.getChild(0) as Element).getChild(0), equalTo("31"))
+        assertThat((root2.getChild(0) as Element).getChild(0), equalTo("32"))
+    }
+
+    private fun parseXml(file: File): Document {
+        return StringReader(String(file.readBytes())).use { reader ->
+            val parser = KXmlParser()
```

Comment:
Ah nice! I'd just meant doing a `contains`, but parsing the XML is definitely more exact. How about combining the common bits here with the parsing done in `StubOpenRosaServer` and pulling it out to some kind of helper function/object? I think we could actually use some of this in real form download code soon as well to avoid needing to use JavaRosa's static parser.